keps/sig-node/3675-resource-plugin-manager/README.md

```diff
@@ -0,0 +1,1261 @@
+3.  Interoperability with Device Plugins, DRA, et cetera. 
+4.	Identify minimal in-cluster operational core (from existing CPU Manager, Memory Manager, Topology Manager)
+5.  Replace CCI policy with a component integrated in None and Static CPU Manager policies
+6.	E2E testing including other components
+
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+* Break any existing use cases for topology manager, memory manager, cpu manager, or 
+device manager.
+* Creating any more latency than there is today for scheduling:  We should be 
+careful not to add any latency into scheduling over what exists today for default behavior.
+* CPU & Memory Management on scheduler side.
+
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+
+### User Stories
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+
+#### Custom workloads, such as Telco/HPC/AI/ML
+
+Custom workloads often have a desire for a mix of types of cores..  For instance, a workload
+should be able to have some number of static cores and some number of shared cores.
+A node should be able to allow both types of cores, without having to have one setting
+or another, and be able to pull from these pulls accordingly.  Additionally, there is
+a need to have some high-priority cores for higher performance and other lower-priority
+cores for other less-sensitive parts of a workloads.  In these use cases, the workloads 
+may also have particular types of NUMA splits required.
+
+#### Power optimization of workloads
+
+Cores should be able to be quickly spun up or down according to resource requirements.
+Additionally, the nodes should be as densely packed as possible regrading the abliity to
+do core use.  There should also be the ability to choose between efficiency cores and
+performance cores within newer architectures, according to workload requirements.
+
+#### Research of new resource management patterns within the cloud
+
+There are a variety of modifiers that can be placed around cores.  Static cores,
+isolated cores, shared cores, efficiency cores, and performance cores are only the
+beginning of unexplored spaces.  Being able to experiment with various patterns in 
+research without having to be an expert in how to modify Kubelet and its multiple 
+internal managers is a decisive benefit to the research community.
```

Comment:
Again, we'll come back to this re scheduling, and does need addressing, potentially with another KEP or within Beta.

Reply:
I am not convinced that we should defer how resources are exposed on the node out of the scope of this KEP to a later stage. This is important for the proper functionality of this feature. We would need the pod lands on a node where it at the very least has the component responsible for allocating resources (CCIDriver) is configured and running AND there are resources to be allocated. There can be exceptional cases where we choose to fail but that should be an exception not a rule.

Reply:
I think that for now just explaining a bit more how you leverage the DRA bits in the flow is sufficient, see [previous comments](https://github.com/kubernetes/enhancements/pull/3853/commits/4d37e0ffadb52b9aaa3d84220f58d58952f1e59f#r1100544852). OTOH the required changes to DRA should be clearly highlighted in the implementation details section

Reply:
I agree with @swatisehgal here. There should be a clear explanation of the design even if it's not going to be implemented in Alpha. So far it looks like we propose not to touch scheduler in Alpha, but somehow assume that this would be enough to implement complex CPU assignments on the node. This approach has little chances to work in my opinion.

Reply:
we added further clarification and an approach which is feasible for alpha stage + post-alpha options